dataprofiler/tests/test_utils_global.py

```diff
@@ -0,0 +1,115 @@
+"""Validates that generator intakes DATAPROFILER_SEED properly."""
+import os
+import unittest
+import unittest.mock
+
+import numpy as np
+from numpy.random import PCG64
+
+from .. import utils_global
+
+
+class TestGetRandomNumberGenerator(unittest.TestCase):
+    """Validates get_random_number_generator() is properly working."""
+
+    @unittest.mock.patch.dict("os.environ", clear=True)
+    def test_return_random_value(self, *mocks):
+        """If DATAPROFILER_SEED not set, test that rng returns different random value."""
+        rng = utils_global.get_random_number_generator()
+        actual_value = rng.integers(0, 1e6)
+        with unittest.mock.patch(
+            "numpy.random.randint",
+        ) as numpy_mock:
+            numpy_mock.return_value = 850624
+            expected_value = np.random.randint(1)
```

Comment:
I think this part might be a little backwards. We don't call randint internally, but are mocking it for at the test level itself. Instead we could just assert the value. is equal to what we expect it \ not. Instead we can mock the creation internally and validate what is called with, e.g. none etc.

Reply:
We can use diff values for settings vs env to validate what it chooses.